// element-plus/lib/components/badge/src/badge.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var badge = require('./badge.js');
var vue = require('vue');
var index = require('../../../hooks/use-namespace/index.js');
var types = require('../../../utils/types.js');
var style = require('../../../utils/dom/style.js');

var _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    name: "ElBadge"
  },
  __name: "badge",
  props: badge.badgeProps,
  setup(__props, { expose: __expose }) {
    const props = __props;
    const ns = index.useNamespace("badge");
    const content = vue.computed(() => {
      if (props.isDot) return "";
      if (types.isNumber(props.value) && types.isNumber(props.max)) {
        return props.max < props.value ? `${props.max}+` : `${props.value}`;
      }
      return `${props.value}`;
    });
    const style$1 = vue.computed(() => {
      var _a;
      return [
        {
          backgroundColor: props.color,
          marginRight: style.addUnit(-props.offset[0]),
          marginTop: style.addUnit(props.offset[1])
        },
        (_a = props.badgeStyle) != null ? _a : {}
      ];
    });
    __expose({
      /** @description badge content */
      content
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock(
        "div",
        {
          class: vue.normalizeClass(vue.unref(ns).b())
        },
        [
          vue.renderSlot(_ctx.$slots, "default"),
          vue.createVNode(vue.Transition, {
            name: `${vue.unref(ns).namespace.value}-zoom-in-center`,
            persisted: ""
          }, {
            default: vue.withCtx(() => [
              vue.withDirectives(vue.createElementVNode(
                "sup",
                {
                  class: vue.normalizeClass([
                    vue.unref(ns).e("content"),
                    vue.unref(ns).em("content", __props.type),
                    vue.unref(ns).is("fixed", !!_ctx.$slots.default),
                    vue.unref(ns).is("dot", __props.isDot),
                    vue.unref(ns).is("hide-zero", !__props.showZero && __props.value === 0),
                    __props.badgeClass
                  ]),
                  style: vue.normalizeStyle(style$1.value)
                },
                [
                  vue.renderSlot(_ctx.$slots, "content", { value: content.value }, () => [
                    vue.createTextVNode(
                      vue.toDisplayString(content.value),
                      1
                      /* TEXT */
                    )
                  ])
                ],
                6
                /* CLASS, STYLE */
              ), [
                [vue.vShow, !__props.hidden && (content.value || __props.isDot || _ctx.$slots.content)]
              ])
            ]),
            _: 3
            /* FORWARDED */
          }, 8, ["name"])
        ],
        2
        /* CLASS */
      );
    };
  }
});

exports.default = _sfc_main;
