// element-plus/es/components/tree-select/src/tree-select.vue2.mjs
import { defineComponent, ref, computed, reactive, onMounted, h } from 'vue';
import { pick } from 'lodash-unified';
import { ElSelect } from '../../select/index.mjs';
import { ElTree } from '../../tree/index.mjs';
import { useSelect } from './select.mjs';
import { useTree } from './tree.mjs';
import CacheOptions from './cache-options.mjs';
import { treeProps } from '../../tree/src/tree.mjs';
import { selectProps } from '../../select/src/select.mjs';

var _sfc_main = defineComponent({
  name: "ElTreeSelect",
  // disable `ElSelect` inherit current attrs
  inheritAttrs: false,
  props: {
    ...selectProps,
    ...treeProps,
    /**
     * @description The cached data of the lazy node, the structure is the same as the data, used to get the label of the unloaded data
     */
    cacheData: {
      type: Array,
      default: () => []
    }
  },
  setup(props, context) {
    const { slots, expose } = context;
    const select = ref();
    const tree = ref();
    const key = computed(() => props.nodeKey || props.valueKey || "value");
    const selectProps2 = useSelect(props, context, { select, tree, key });
    const { cacheOptions, ...treeProps2 } = useTree(props, context, {
      select,
      tree,
      key
    });
    const methods = reactive({});
    expose(methods);
    onMounted(() => {
      Object.assign(methods, {
        //TODO: let only tree and select in 3.0
        ...pick(tree.value, [
          "filter",
          "updateKeyChildren",
          "getCheckedNodes",
          "setCheckedNodes",
          "getCheckedKeys",
          "setCheckedKeys",
          "setChecked",
          "getHalfCheckedNodes",
          "getHalfCheckedKeys",
          "getCurrentKey",
          "getCurrentNode",
          "setCurrentKey",
          "setCurrentNode",
          "getNode",
          "remove",
          "append",
          "insertBefore",
          "insertAfter"
        ]),
        ...pick(select.value, ["focus", "blur", "selectedLabel"]),
        treeRef: tree.value,
        selectRef: select.value
      });
    });
    return () => h(
      ElSelect,
      /**
       * 1. The `props` is processed into `Refs`, but `v-bind` and
       * render function props cannot read `Refs`, so use `reactive`
       * unwrap the `Refs` and keep reactive.
       * 2. The keyword `ref` requires `Ref`, but `reactive` broke it,
       * so use function.
       */
      reactive({
        ...selectProps2,
        ref: (ref2) => select.value = ref2
      }),
      {
        ...slots,
        default: () => [
          h(CacheOptions, { data: cacheOptions.value }),
          h(
            ElTree,
            reactive({
              ...treeProps2,
              ref: (ref2) => tree.value = ref2
            })
          )
        ]
      }
    );
  }
});

export { _sfc_main as default };
